natiami/assignment4/FindIslands.cpp

```diff
@@ -0,0 +1,45 @@
+#include"FindIslands.h"
+
+FindIslands::FindIslands(){}
+
+FindIslands::FindIslands(int nr_row, int nr_column, vector<vector<bool> >& mapOfTiles) : nr_row(nr_row), nr_column(nr_column), mapOfTiles(mapOfTiles){
+}
+
+bool FindIslands::checkBounds( int i, int j){
+	return i >= 0 && i < nr_row && j >= 0 && j < nr_column;
+}
+
+void FindIslands::visitOneIsland(vector<vector<bool> >& visited, int i, int j){
+	visited[i][j] = true;
+	vector<pair<int, int> > moves{ { 0, 1 }, { 1, 0 }, { 0, -1 }, { -1, 0 } };
+	int row, column;
+
+	for (size_t k = 0; k < moves.size(); k++)
+	{
+		row = i + moves[k].first;
+		column = j + moves[k].second;
+		if (checkBounds(row, column)){
+			if (!visited[row][column] && mapOfTiles[row][column])

```

Comment:
The line
```C++
if (!visited[row][column] && mapOfTiles[row][column])
```
is duplicated both in here and in `findNumberOfIsland`.

To avoid this duplication you can move these checks it to the beginning of `visitOneIsland`:
```C++
void FindIslands::visitOneIsland(vector<vector<bool> >& visited, int i, int j){
    if (!checkBounds(i,j) || !mapOfTiles[i][j] || visited[i][j]) {
        return;
    }
    ...
    for (...) {
        row=...;
        col=...;
        visitOneIsland(visited, row, col);  // You can now call this directly from the loop without surrounding with checks.
    }
}
```
Same below - `visitOneIsland` can be called directly from the loop in `findNumberOfIslands`.